paimon-core/src/test/java/org/apache/paimon/iceberg/IcebergCompatibilityTest.java

```diff
@@ -160,6 +162,52 @@ public void testPartitionedPrimaryKeyTable() throws Exception {
                 r -> String.format("%d|%d", r.get(3, Integer.class), r.get(4, Long.class)));
     }
 
+    @Test
+    public void testPartitionedPrimaryKeyTableTimestamp() throws Exception {
```

Comment:
This random test is kind of useless because it didn't catch any of your bugs I mentioned above. Could you think of a more efficient way to test your code?